keras/activations.py

```diff
@@ -4,13 +4,25 @@
 from .utils.generic_utils import deserialize_keras_object
 
 
-def softmax(x):
+def softmax(x, axis=-1):
+    """Softmax activation function.
+
+    # Arguments
+        x : tensor
+        axis: integer, defines along which axis the normalization is applied
+
+    # Returns
+        Tensor to which the softmax activation function is applied
+
+    # Raises
+        ValueError: In case dim(x) < 2.
+    """
     ndim = K.ndim(x)
     if ndim == 2:
         return K.softmax(x)
-    elif ndim == 3:
-        e = K.exp(x - K.max(x, axis=-1, keepdims=True))
-        s = K.sum(e, axis=-1, keepdims=True)
+    elif ndim > 2:
+        e = K.exp(x - K.max(x, axis=axis, keepdims=True))
+        s = K.sum(e, axis=axis, keepdims=True)
         return e / s
     else:
         raise ValueError('Cannot apply softmax to a tensor '
```

Comment:
There is only one case, `ndim < 2`, to raise `ValueError` and error message should be fixed.